### PR TITLE
Value object pattern #349

### PR DESCRIPTION
**Pull request by JuhoKang**
Implementation for the Value Object pattern.
I had some difficulties to show all of the VALJO things so just stopped in the middle and just implemented some "must do" things in the code.

### CODE REVIEW

**Comment by iluwatar** [comment]
Could you catch up the branch from the master, it has conflicts.

File: front-controller/index.md [hunk 1]
```diff
@@ -3,14 +3,14 @@ layout: pattern
 title: Front Controller
 folder: front-controller
 permalink: /patterns/front-controller/
-categories: Presentation Tier
+categories: Creational
```
Comment (iluwatar) [inline front-controller/index.md]: Did you mean to change this file?

File: front-controller/index.md [hunk 2]
```diff
@@ -3,14 +3,14 @@ layout: pattern
...
 tags:
  - Java
  - Difficulty-Intermediate
 ---
 
 ## Intent
-Introduce a common handler for all requests for a web site. This
+Introduce how to create objects which follow value semantics rather than reference semantics.
```
Comment (iluwatar) [inline front-controller/index.md]: Did you mean to change this?

File: value-object/src/main/java/com/iluwatar/value/object/App.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.iluwatar.value.object;
+
+/**
+ * App Class.
```
Comment (iluwatar) [inline value-object/src/main/java/com/iluwatar/value/object/App.java]: Please describe the pattern and the example here. For reference, see other pattern examples.

**Comment by JuhoKang** [comment]
I'll catch up changes, add description as you asked. Sorry for not checking trivial mistakes making other files to change. I'll comment on this thread when done.

File: pom.xml [hunk 4]
```diff
@@ -354,7 +355,7 @@
 					  <excludeFromFailureFile>exclude-pmd.properties</excludeFromFailureFile>
 				  </configuration>
                 </execution>
-              </executions>
```
Comment (JuhoKang) [inline pom.xml]: I did no edits except adding the \<module>value-object\</module> but github says those spaces are edited. I don't know why

Comment (JuhoKang) [inline pom.xml]: I did no edits except adding the \<module>value-object\</module> but github says those spaces are edited. I don't know why

Comment (JuhoKang) [inline pom.xml]: I don't know why these blanks exist and make differences in this file. I only added one line \<module>value-object\</module>.

**Comment by JuhoKang** [comment]
I think the fixes are done. please review. #362

**Comment by iluwatar** [comment]
Thanks @JuhoKang this looks good!

**Comment by iluwatar** [comment]
@all-contributors please add @JuhoKang for code

**Pull request merged**